awscrt/auth.py

```diff
@@ -255,6 +255,32 @@ def new_chain(cls, providers):
         binding = _awscrt.credentials_provider_new_chain(providers)
         return cls(binding)
 
+    @classmethod
+    def new_python(cls, py_provider):
```

Comment:
naming: not a fan of using the word "python" to distinguish something ... in python. I would just name this `def new_delegate(get_credentials)`